wire-ios-data-model/Source/MLS/OneOnOne/OneOnOneMigrator.swift

```diff
@@ -147,13 +148,19 @@ public struct OneOnOneMigrator: OneOnOneMigratorInterface {
             ) else {
                 throw MigrateMLSOneOnOneConversationError.failedToActivateConversation
             }
-            guard !mlsConversation.migratedToMLS else {
-                throw MigrateMLSOneOnOneConversationError.alreadyMigrated
-            }
+
             guard let otherUser = ZMUser.fetch(with: userID, in: context) else {
                 throw MigrateMLSOneOnOneConversationError.failedToActivateConversation
             }
 
+            guard !(mlsConversation.migratedToMLS && otherUser.oneOnOneConversation == mlsConversation) else {
+                throw MigrateMLSOneOnOneConversationError.alreadyMigrated
+            }
```

Comment:
todo: please report this logic to WireDomain's OneOnOneResolver too

Reply:
doen